lib/lltz_michelson/lltz_michelson.ml

```diff
@@ -362,8 +376,50 @@ let rec compile : LLTZ.E.t -> Instruction.t =
                   update_n num ~length:(List.nth_exn lengths i)))
          in
          seq ([ compile tuple ] @ gets @ [ compile update ] @ updates)
-       | Inj (path, expr) -> assert false
-       | Match (subject, cases) -> assert false
+       | Inj (context, expr) -> 
+        (*crete or type*)
+        (*middle type*)
+        (match context with
+        | LLTZ.R.Context.Hole ty -> 
+          let mid_ty = convert_type ty in
+          seq [ compile expr; ]
+        | LLTZ.R.Context.Node (left_val, mid ,right_val) ->  
+          let right_ty = Type.ors (List.map ~f:compile_row_types_for_or right_val) in
+          let mid_ty = Type.ors (List.map ~f:compile_row_types_for_or [mid]) in
+          (*go(fold) through all elements in left (<- direction) and righ-comb iteratively merge them
+        into a larger or type, record each intermediate merge*)
+          let right_instrs_types = (List.fold_right left_val
+          ~f:(fun x lst -> 
+            match lst with
+            | hd::tl -> Type.ors ([compile_row_types_for_or x; hd])::(hd::tl)
+            | [] -> raise_s [%message "Empty list"]
+          )
+          ~init:([Type.ors ([mid_ty; right_ty])])) in
+          seq ([
+            compile expr;
+            (* Left *)
+            left mid_ty;
+            ]
+            (* Rights - traverses all right_instrs_types in reverse order except last and makes right*)
+            @ (
+              if List.length right_instrs_types = 0 then []
```

Comment:
This is inefficient:
```
match right_instrs_types with
  | [] -> []
  | _ -> List.map (...) ~f:(...) 
  ```
and use a "let ... in .." instead of a `@ (... big stuff here ...)`.